### PR TITLE
feat: Overhauls the builder for better OpenShift Enterprise support.

### PR DESCRIPTION
**Pull request by jeremyzahner**
- Adds images for 1.5 and 1.6
- Improves compatibility to different build environments.
- Adds new improved image stream declaration.

### CODE REVIEW

File: 1.2/Dockerfile [hunk 1]
```diff
@@ -12,7 +12,13 @@ ENV NPM_VERSION ${NPM_VERSION}
 LABEL io.k8s.description="Platform for building and running a phoenix app" \
       io.k8s.display-name="build-phoenix" \
       io.openshift.expose-services="4000:http" \
-      io.openshift.tags="builder,elixir,phoenix"
+      io.openshift.tags="builder,elixir,phoenix" \
+      com.redhat.dev-mode="DEV_MODE:false" \
+      com.redhat.deployments-dir="/opt/app-root/src" \
+      com.redhat.dev-mode.port="DEBUG_PORT:5858" \
```
Comment (maennchen) [inline 1.2/Dockerfile]: What is the purpose of the debug port? There is nothing listening on that port...

File: 1.2/Dockerfile [hunk 2]
```diff
@@ -12,7 +12,13 @@ ENV NPM_VERSION ${NPM_VERSION}
 LABEL io.k8s.description="Platform for building and running a phoenix app" \
       io.k8s.display-name="build-phoenix" \
       io.openshift.expose-services="4000:http" \
-      io.openshift.tags="builder,elixir,phoenix"
+      io.openshift.tags="builder,elixir,phoenix" \
+      com.redhat.dev-mode="DEV_MODE:false" \
```
Comment (maennchen) [inline 1.2/Dockerfile]: What does that label do?

File: 1.3/Dockerfile [hunk 3]
```diff
@@ -12,7 +12,13 @@ ENV NPM_VERSION ${NPM_VERSION}
 LABEL io.k8s.description="Platform for building and running a phoenix app" \
       io.k8s.display-name="build-phoenix" \
       io.openshift.expose-services="4000:http" \
-      io.openshift.tags="builder,elixir,phoenix"
+      io.openshift.tags="builder,elixir,phoenix" \
+      com.redhat.dev-mode="DEV_MODE:false" \
+      com.redhat.deployments-dir="/opt/app-root/src" \
+      com.redhat.dev-mode.port="DEBUG_PORT:5858" \
```
Comment (maennchen) [inline 1.3/Dockerfile]: Same as `1.2`

File: 1.4/Dockerfile [hunk 4]
```diff
@@ -12,7 +12,13 @@ ENV NPM_VERSION ${NPM_VERSION}
 LABEL io.k8s.description="Platform for building and running a phoenix app" \
       io.k8s.display-name="build-phoenix" \
       io.openshift.expose-services="4000:http" \
-      io.openshift.tags="builder,elixir,phoenix"
+      io.openshift.tags="builder,elixir,phoenix" \
+      com.redhat.dev-mode="DEV_MODE:false" \
+      com.redhat.deployments-dir="/opt/app-root/src" \
+      com.redhat.dev-mode.port="DEBUG_PORT:5858" \
```
Comment (maennchen) [inline 1.4/Dockerfile]: Same as `1.2`

File: 1.5/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,117 @@
+FROM openshift/base-centos7
+
+ARG ERLANG_VERSION=20.0
```
Comment (maennchen) [inline 1.5/Dockerfile]: That's an old version

File: 1.5/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+ENV NODE_VERSION ${NODE_VERSION}
+ENV NPM_VERSION ${NPM_VERSION}
+
+LABEL io.k8s.description="Platform for building and running a phoenix app" \
+      io.k8s.display-name="build-phoenix" \
+      io.openshift.expose-services="4000:http" \
+      io.openshift.tags="builder,elixir,phoenix" \
+      com.redhat.dev-mode="DEV_MODE:false" \
+      com.redhat.deployments-dir="/opt/app-root/src" \
+      com.redhat.dev-mode.port="DEBUG_PORT:5858" \
```
Comment (maennchen) [inline 1.5/Dockerfile]: Same as `1.2`

File: 1.6/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,117 @@
+FROM openshift/base-centos7
+
+ARG ERLANG_VERSION=20.0
```
Comment (maennchen) [inline 1.6/Dockerfile]: That's an old version

File: 1.6/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+ENV NODE_VERSION ${NODE_VERSION}
+ENV NPM_VERSION ${NPM_VERSION}
+
+LABEL io.k8s.description="Platform for building and running a phoenix app" \
+      io.k8s.display-name="build-phoenix" \
+      io.openshift.expose-services="4000:http" \
+      io.openshift.tags="builder,elixir,phoenix" \
+      com.redhat.dev-mode="DEV_MODE:false" \
+      com.redhat.deployments-dir="/opt/app-root/src" \
+      com.redhat.dev-mode.port="DEBUG_PORT:5858" \
```
Comment (maennchen) [inline 1.6/Dockerfile]: Same as `1.2`

**Review by maennchen — CHANGES_REQUESTED** [review]

File: 1.2/Dockerfile (hunk 1 above)
Comment (jeremyzahner) [reply 1.2/Dockerfile]: Removed in 551f804, I did accidentally copy that.

File: 1.2/Dockerfile (hunk 2 above)
Comment (jeremyzahner) [reply 1.2/Dockerfile]: Removed in 551f804, I did accidentally copy that.

File: 1.3/Dockerfile (hunk 3 above)
Comment (jeremyzahner) [reply 1.3/Dockerfile]: Removed in 551f804, I did accidentally copy that.

File: 1.4/Dockerfile (hunk 4 above)
Comment (jeremyzahner) [reply 1.4/Dockerfile]: Removed in 551f804, I did accidentally copy that.

File: 1.5/Dockerfile (hunk 5 above)
Comment (jeremyzahner) [reply 1.5/Dockerfile]: @maennchen What version would you prefer for 1.5?

File: 1.6/Dockerfile (hunk 7 above)
Comment (jeremyzahner) [reply 1.6/Dockerfile]: @maennchen What version would you prefer for 1.6?

File: 1.5/Dockerfile (hunk 6 above)
Comment (jeremyzahner) [reply 1.5/Dockerfile]: Removed in 551f804, I did accidentally copy that.

File: 1.6/Dockerfile (hunk 8 above)
Comment (jeremyzahner) [reply 1.6/Dockerfile]: Removed in 551f804, I did accidentally copy that.

**Comment by jeremyzahner** [comment]
@maennchen I've just copied the test-app in the tests folder for 1.5 / 1.6. Should we change something there as well?

File: 1.5/Dockerfile (hunk 5 above)
Comment (maennchen) [reply 1.5/Dockerfile]: The newest stable release, `20.3`

File: 1.6/Dockerfile (hunk 7 above)
Comment (maennchen) [reply 1.6/Dockerfile]: The newest stable release, `20.3`

File: 1.5/Dockerfile (hunk 5 above)
Comment (jeremyzahner) [reply 1.5/Dockerfile]: Done in 21ce772

File: 1.6/Dockerfile (hunk 7 above)
Comment (jeremyzahner) [reply 1.6/Dockerfile]: Done in 21ce772

**Pull request merged**